### PR TITLE
[CI-3413] Add analytics tags param

### PR DESCRIPTION
**Pull request by stanlp1**
Adds two things.

1. defaultAnalyticsTags to the config.
2. analyticsTags param to some tracking functions.

defaultAnalyticsTags will be passed with every request if exists, will be merged with analyticsTags if exists.

### CODE REVIEW

File: AutocompleteClient/FW/Config/ConstructorIOConfig.swift [hunk 1]
```diff
@@ -1,6 +1,4 @@
-//
-//  ConstructorIOConfig.swift
-//  Constructor.io
+//config
```
Comment (mocca102) [inline AutocompleteClient/FW/Config/ConstructorIOConfig.swift:1]: Why did we do this?

**Comment by stanlp1** [comment]
> Are we intentionally not adding the analytics tags to the search events?

AFAIK, search is on v1 GET events, don't support the analytics tags yet is the only reason I didn't add them.

File: AutocompleteClient/FW/Config/ConstructorIOConfig.swift (hunk 1 above)
Comment (stanlp1) [reply AutocompleteClient/FW/Config/ConstructorIOConfig.swift:1]: 🤔  Not sure why that happened. Will revert.

**Review by mocca102 — APPROVED** [review]
LGTM!

**Comment by mocca102** [comment]
@stanlp1 There are some conflicts with master please make sure to resolve them before merging

**Pull request merged**